### PR TITLE
tools: support asan

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -213,6 +213,8 @@ failpoint-disable: tools/bin/failpoint-ctl
 	@$(FAILPOINT_DISABLE)
 
 tools/bin/ut: tools/check/ut.go
+	sudo  yum install -y centos-release-scl; \
+	sudo yum install -y devtoolset-8-libasan-devel; \
```
Comment (xhebox) [inline Makefile:217]: I suggest removing this. Not every one is using yum.

File: tools/check/ut.go [hunk 2]
```diff
@@ -795,9 +795,9 @@ func buildTestBinary(pkg string) error {
 	// go test -c
 	var cmd *exec.Cmd
 	if coverprofile != "" {
-		cmd = exec.Command("go", "test", "-c", "-cover", "-vet", "off", "-o", testFileName(pkg))
+		cmd = exec.Command("go", "test", "-asan", "-c", "-cover", "-vet", "off", "-o", testFileName(pkg))
```
Comment (xhebox) [inline tools/check/ut.go:798]: I think this can be done in bazel now.

File: Makefile (hunk 1 above)
Comment (hawkingrei) [reply Makefile:217]: Now we can install those in our docker image. so it is useless.

**Comment by hawkingrei** [comment]
Let me try to add asan into bazel.

**Pull request closed without merging**